storage/cache/subtree_cache.go

```diff
@@ -189,10 +189,11 @@ func (s *SubtreeCache) SetNodeHash(id storage.NodeID, h trillian.Hash, getSubtre
 }
 
 // Flush causes the cache to write all dirty Subtrees back to storage.
-func (s *SubtreeCache) Flush(setSubtree SetSubtreeFunc) error {
+func (s *SubtreeCache) Flush(setSubtrees SetSubtreesFunc) error {
 	s.mutex.RLock()
 	defer s.mutex.RUnlock()
 
+	treesToWrite := make([]*storage.SubtreeProto, 0, len(s.dirtyPrefixes))
```

Comment:
Seems possible that we might have a large number of dirty prefixes? I don't know if this is a real problem